gcc/d/expr.cc

```diff
@@ -117,15 +117,15 @@ class ExprVisitor : public Visitor
       {
 	// Convert arrays to D array types.
 	tree t1 = d_array_convert(e->e1);
-	t1 = maybe_make_temp(t1);
 	tree t2 = d_array_convert(e->e2);
-	t2 = maybe_make_temp(t2);
 	this->result_ = build2(code, build_ctype(e->type), t1, t2);
```

Comment:
I actually wanted to replace the `result = ...` by this:
```d
	this->result_ = d_convert(build_ctype(e->type),
				  build_boolop(code, t1, t2));
```
But forgot to copy this over from the ARM machine...